src/algebra/ordered_monoid.lean

```diff
@@ -380,21 +374,48 @@ lemma add_eq_coe [has_add α] : ∀ {a b : with_top α} {c : α},
 | (some a) (some b) c :=
     by simp only [some_eq_coe, ← coe_add, coe_eq_coe, exists_and_distrib_left, exists_eq_left]
 
+instance [add_semigroup α] : add_semigroup (with_top α) :=
+{ add_assoc :=
+  begin
+    refine with_top.rec_top_coe _ _,
```

Comment:
can you make sure to add braces when you have multiple goals, and also to avoid non-terminal simps (it is hard to tell from the proof structure now if your simps are terminal or not :-). Same thing for the next proof.

Reply:
I was hoping to get comments on this =) I'd like to phrase this as the brute-force case bash that it is. We can see that the steps are very much similar. Is there some `try` or `repeat` tactic that can golf this further?